packages/core/src/browser/common-frontend-contribution.ts

```diff
@@ -117,216 +117,218 @@ export namespace CommonCommands {
         id: 'core.copy.path',
         label: 'Copy Path'
     });
 
-    export const UNDO = Command.toDefaultLocalizedCommand({
+    export const UNDO = Command.toDefaultLocalizedCommand<[], void>({
         id: 'core.undo',
         label: 'Undo'
     });
-    export const REDO = Command.toDefaultLocalizedCommand({
+    export const REDO = Command.toDefaultLocalizedCommand<[], void>({
         id: 'core.redo',
         label: 'Redo'
     });
-    export const SELECT_ALL = Command.toDefaultLocalizedCommand({
+    export const SELECT_ALL = Command.toDefaultLocalizedCommand<[], void>({
         id: 'core.selectAll',
         label: 'Select All'
     });
 
+    // TODO: type command
     export const FIND = Command.toDefaultLocalizedCommand({
         id: 'core.find',
         label: 'Find'
     });
+    // TODO: type command
     export const REPLACE = Command.toDefaultLocalizedCommand({
         id: 'core.replace',
         label: 'Replace'
     });
 
-    export const NEXT_TAB = Command.toDefaultLocalizedCommand({
+    export const NEXT_TAB = Command.toDefaultLocalizedCommand<[], void>({
         id: 'core.nextTab',
         category: VIEW_CATEGORY,
         label: 'Show Next Tab'
     });
-    export const PREVIOUS_TAB = Command.toDefaultLocalizedCommand({
+    export const PREVIOUS_TAB = Command.toDefaultLocalizedCommand<[], void>({
         id: 'core.previousTab',
         category: VIEW_CATEGORY,
         label: 'Show Previous Tab'
     });
-    export const NEXT_TAB_IN_GROUP = Command.toLocalizedCommand({
+    export const NEXT_TAB_IN_GROUP = Command.toLocalizedCommand<[], void>({
         id: 'core.nextTabInGroup',
         category: VIEW_CATEGORY,
         label: 'Switch to Next Tab in Group'
     }, 'theia/core/common/showNextTabInGroup', VIEW_CATEGORY_KEY);
-    export const PREVIOUS_TAB_IN_GROUP = Command.toLocalizedCommand({
+    export const PREVIOUS_TAB_IN_GROUP = Command.toLocalizedCommand<[], void>({
         id: 'core.previousTabInGroup',
         category: VIEW_CATEGORY,
         label: 'Switch to Previous Tab in Group'
     }, 'theia/core/common/showPreviousTabInGroup', VIEW_CATEGORY_KEY);
-    export const NEXT_TAB_GROUP = Command.toLocalizedCommand({
+    export const NEXT_TAB_GROUP = Command.toLocalizedCommand<[], void>({
         id: 'core.nextTabGroup',
         category: VIEW_CATEGORY,
         label: 'Switch to Next Tab Group'
     }, 'theia/core/common/showNextTabGroup', VIEW_CATEGORY_KEY);
-    export const PREVIOUS_TAB_GROUP = Command.toLocalizedCommand({
+    export const PREVIOUS_TAB_GROUP = Command.toLocalizedCommand<[], void>({
         id: 'core.previousTabBar',
         category: VIEW_CATEGORY,
         label: 'Switch to Previous Tab Group'
     }, 'theia/core/common/showPreviousTabGroup', VIEW_CATEGORY_KEY);
-    export const CLOSE_TAB = Command.toLocalizedCommand({
+    export const CLOSE_TAB = Command.toLocalizedCommand<[event: Event], void>({
```

Comment:
A number of labeled tuples in this PR, which we know could be breaking for downstream projects who don't keep up to date with TS. Are they really adding value?

Reply:
Named tuples are already in the code base (found at least one occurrence):

https://github.com/eclipse-theia/theia/blame/a58e2f474c168d322a950fcef26bc4c95472f226/packages/core/src/common/menu/menu-adapter.ts#L22